HelloWorld app for Hydra

tahir24434 opened the pull request:
Signed-off-by: tahir24434 tahir.rauf1@gmail.com

tahir24434 committed:
WIP: HelloWorld app for Hydra

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

tahir24434 committed:
Merge branch 'master' into master

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

sushilks commented on src/main/python/hydra/helloworldtest/__init__.py:
```diff
@@ -0,0 +1 @@
+__author__ = 'AbdullahS'
```
why ?

sushilks commented on src/main/scripts/hydra:
```diff
@@ -17,6 +17,7 @@ classPath = {
     'rmq' : 'hydra.rmqtest.runtest.RunTest',
     'rmqpub' : 'hydra.rmqtest.rmq_pub.run',
     'rmqfixed': 'hydra.rmqtest.fixed_test.run',
+    'hw': 'hydra.helloworldtest.runtest.RunTest',
```
seems fine.

sushilks replied:
If you are getting error likely the files for hellowwordtest are not installed.

tahir24434 committed:
Helloworld app for Hydra: fixes

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

tahir24434 committed:
Raising a descriptive exception about invalid app_id

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

kratos7 commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,84 @@
+    def __init__(self):
+        self.config = ConfigParser()  # Object contains 'configurations' parsed from hydra.ini config file
+        RunTestBase.__init__(self, 'HelloWorld', None, None, startappserver=True, mock=False)
+        self.hw_server_app_id = super(HW, self).format_appname("/hw-server")
```
recommended is to use self.method  , super is redundant here.

tahir24434 replied:
Done.

tahir24434 committed:
Running integration tests

Integration tests will run as part of "pyb install -x run_unit_tests" command.

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

sushilks commented on build.py:
```diff
@@ -6,7 +6,7 @@
 use_plugin("filter_resources")
 use_plugin("python.unittest")
-#use_plugin("python.integrationtest")
+use_plugin("python.integrationtest")
```
probably need travis file updates to exclude from running integration test

tahir24434 replied:
Done.

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,84 @@
+class HW(RunTestBase):
+    def __init__(self):
+        self.config = ConfigParser()  # Object contains 'configurations' parsed from hydra.ini config file
+        RunTestBase.__init__(self, 'HelloWorld', None, None, startappserver=True, mock=False)
```
If defaults for mock is false and startappserver is true then those should be omitted.

tahir24434 replied:
Done

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,84 @@
+
+    def run_test(self):
+        # Get Mesos/Marathon client
+        super(HW, self).start_init()
```
self.start_init()

tahir24434 replied:
Done.

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,84 @@
+
+    def launch_hw_server(self):
+        print ("Launching the HelloWorld server app")
+        constraints = [self.app_constraints(field='hostname', operator='UNIQUE')]
```
why is UNIQUE required ?

tahir24434 replied:
I added it as I wanted to run the publisher only on slave-set1_0. Why I tried to launch it only on slave-set1_0? I just wanted to showcase how one can add constraints using Hydra.
I can remove it.

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,84 @@
+        constraints = [self.app_constraints(field='hostname', operator='UNIQUE')]
+
+        # Use cluster0 for launching the hw_server
+        if 0 in self.mesos_cluster:
```
This is not needed as well

tahir24434 replied:
Done

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,84 @@
+            # field: slave_id, operator: CLUSTER, value: slave-set1_0
+            constraints.append(self.app_constraints(field=self.mesos_cluster[0]['cat'],
+                                                    operator='CLUSTER', value=self.mesos_cluster[0]['match']))
+        self.create_hydra_app(name=self.hw_server_app_id, app_path='hydra.helloworldtest.hw_server.run',
```
Can you change the app from hydra app to running a shell command ?

tahir24434 replied:
LIB-185

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,84 @@
+        print ("Launching the HelloWorld client app")
+        constraints = []
+        # Use cluster 1 for launching the SUB
+        if 1 in self.mesos_cluster:
```
Also not needed

tahir24434 replied:
Done.

tahir24434 committed:
addressing comments

Signed-off-by: tahir24434 <tahir.rauf1@gmail.com>

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,81 @@
+        constraints = []
+        # Use cluster0 for launching the hw_server
+        # field: slave_id, operator: CLUSTER, value: slave-set1_0
+        constraints.append(self.app_constraints(field=self.mesos_cluster[0]['cat'],
```
Not needed.

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,81 @@
+        print ("Launching the HelloWorld client app")
+        constraints = []
+        # Use cluster 1 for launching the SUB
+        constraints.append(self.app_constraints(field=self.mesos_cluster[1]['cat'],
```
Not needed

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,81 @@
+        # field: slave_id, operator: CLUSTER, value: slave-set1_0
+        constraints.append(self.app_constraints(field=self.mesos_cluster[0]['cat'],
+                                                operator='CLUSTER', value=self.mesos_cluster[0]['match']))
+        self.create_hydra_app(name=self.hw_server_app_id, app_path='hydra.helloworldtest.hw_server.run',
```
Can you change the app to a shell launched app?

sushilks commented on src/main/python/hydra/helloworldtest/runtest.py:
```diff
@@ -0,0 +1,81 @@
+        # Launch HelloWorld server
+        self.launch_hw_server()
+        # Launch HelloWorld client
+        self.launch_hw_client()
```
There has to be some check's after the app's are launched. 
i.e. you can ask the one guy to send the "hello world" few times 
then you can ask the other guy how many times did he receive the hello world 
this will make sure the test is running.

tahir24434 replied:
Good suggestion. I will do it sometime this week. To keep track of it, I have added ticket  LIB-186 .
